my-app/src/components/Login.js

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect } from "react";
+import PopupWithForm from "./PopupWithForm";
+import { useNavigate } from "react-router-dom";
+
+function Login(props) {
+  const [email, setEmail] = useState("");
+  const [emailError, setEmailError] = useState("");
+  function handleChangeEmail(e) {
+    setEmail(e.target.value);
+  }
+  
+  const [password, setPassword] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+  function handleChangePassword(e) {
+    setPassword(e.target.value);
+  }
+
+  let navigate = useNavigate();
+
+  const isEmailValid = props.isInputValid(email, "email");
+  const isPasswordValid = props.isInputValid(password, "password");
+  const areInputsValid = isEmailValid && isPasswordValid;
+
+  useEffect(() => {
+    if (isEmailValid) {
+      setEmailError("");
+    } else {
+      setEmailError("Invalid email format");
+    }
+
+    if (isPasswordValid) {
+      setPasswordError("");
+    } else {
+      setPasswordError("Password must be at least 3 characters");
+    }
+  }, [isEmailValid, isPasswordValid]);
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    console.log("send request");
+    props.onSignIn({
+      password,
+      email,
+    });
+    setTimeout(() => {
+      navigate("/saved-news");
+    }, 1000);
+  }
+
+  return (
+    <>
```

Comment:
`<>` empty tags are not needed if you have a common tag that wraps the whole markup of the component